pkg/virt-controller/watch/vmi.go

```diff
@@ -1139,24 +1156,24 @@ func (c *VMIController) hotplugPodsReady(vmi *virtv1.VirtualMachineInstance, vir
 	return true, nil
 }
 
-func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod, dataVolumes []*cdiv1.DataVolume) syncError {
+func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod, dataVolumes []*cdiv1.DataVolume) (*k8sv1.Pod, syncError) {
```

Comment:
Why do you need to return `Pod`?

Reply:
Because AFAIK, the intention in [line](https://github.com/kubevirt/kubevirt/pull/10961/files#diff-16487fff595728eb5eeec01225a2203756ecad25d90e90d5feb13a9ff08df7ffR1232) 


```go
pod, err := c.clientset.CoreV1().Pods(vmi.GetNamespace()).Create(context.Background(), templatePod, v1.CreateOptions{})
		//this will fail since pod is nil
		if k8serrors.IsForbidden(err) && strings.Contains(err.Error(), "violates PodSecurity") {
			psaErr := fmt.Errorf("failed to create pod for vmi %s/%s, it needs a privileged namespace to run: %w", vmi.GetNamespace(), vmi.GetName(), err)
			c.recorder.Eventf(vmi, k8sv1.EventTypeWarning, FailedCreatePodReason, failedToRenderLaunchManifestErrFormat, psaErr)
			return &syncErrorImpl{psaErr, FailedCreatePodReason}
			return pod, &syncErrorImpl{psaErr, FailedCreatePodReason}
		}
```

is if the pod is not found, we need to create it, assign it to `pod` variable and reflect the change outside the `sync` function in order to update the vmi state later on. So, if I'm right, please correct me if I'm wrong, as it is the `pod` variable change won't be seen outside the sync function.

Reply:
It is not required because the informer will enqueue the VMI once it observes that we have created the Pod.

Reply:
Right, but it takes time, which is what I'm trying to reduce here. For putting in context, as it is, the launcher_respawn time is nearly 1 second, and without retuning the pod the launcher_respawn time is 20 seconds. I didn't have the chance to try out on much more complex scenarios. However, I expect this time to be much higher.

Reply:
This should not be a problem. An informer should observe changes and it should trigger the next resync fast enough for us to observe. Could we retest and investigate this if you observe this again?

Reply:
Currently there is the chance for a gap between creation of the Pod and reenqueing it for another run to update the VM's status. The informer should be pretty fast, but there is a slight chance that the VM's status and the reality (Pod already created) become out of sync. IMO this is a neat and simple way to create the Pod and update the VM's status in one go.

Reply:
I've carried out a much more detailed experimentation. The result is this change does not have a meaningful impact on the final recovery time. Therefore, you are right, and this change has been reverted. Nevertheless, IMHO, we should consider this change in the future.